modules/tile-converter/src/types.ts

```diff
@@ -136,3 +168,200 @@ type FieldInfo = {
   isEditable: boolean;
   label: string;
 };
+
+// Spec - https://github.com/Esri/i3s-spec/blob/master/docs/1.7/3DNodeIndexDocument.cmn.md
+export type Node3DIndexDocument = {
+  id: string;
+  version?: string;
+  path?: string;
+  level?: number;
+  mbs?: Mbs;
+  obb?: Obb;
+  lodSelection?: LodSelection;
+  children?: NodeReference[];
+  neighbors?: NodeReference[];
+  parentNode?: NodeReference;
+  sharedResource?: Resource;
+  featureData?: Resource[];
+  geometryData?: Resource[];
+  textureData?: Resource[];
+  attributeData?: Resource[];
+  created?: string;
+  expires?: string;
+};
+
+export type LodSelection = {
+  metricType?: string;
+  maxError: number;
+};
+
+export type NodeReference = {
+  id: string;
+  version?: string;
+  mbs?: Mbs;
+  obb?: Obb;
+  href?: string;
+};
+
+export type Resource = {
+  href: string;
+  layerContent?: string[];
+  featureRange?: number[];
+  multiTextureBundle?: string;
+  vertexElements?: number[];
+  faceElements?: number[];
+  nodePath?: string;
+};
+
+export type I3SGeometry = {
+  geometry: ArrayBuffer | null;
+  compressedGeometry?: ArrayBuffer | null;
+  texture: any | null;
+  sharedResources: SharedResources | null;
+  meshMaterial?: GLTFMaterial | null;
+  vertexCount: number | null;
+  attributes: any | null;
+  featureCount: number | null;
+  geometryBuffer?: ArrayBuffer;
+};
+
+export type MaxScreenThresholdSQ = {
+  maxError: number;
+};
+
+export type NodeInPage = {
+  index?: number;
+  lodThreshold: number;
+  obb: Obb;
+  children: any[];
+  mesh: Mesh;
+};
+
+type Mesh = {
+  geometry: {
+    definition: number;
+  };
+  attribute: any;
+};
+
+export type SharedResources = {
+  materialDefinitions?: GLTFMaterial[];
+  textureDefinitions?: TextureDefinitionInfo[];
+  nodePath: string;
+};
+
+type TextureDefinitionInfo = {
+  encoding: string[];
+  wrap?: string[];
+  atlas?: boolean;
+  uvSet?: string;
+  channels?: 'rbg' | 'rgba' | string;
+  images: TextureImage[];
+};
+
+export type TextureImage = {
+  id: string;
+  size?: number;
+  pixelInWorldUnits?: number;
+  href?: string[];
+  byteOffset?: string[];
+  length?: number[];
+  mimeType?: string;
+  bufferView?: {
+    data: ArrayBuffer;
+  };
+  image?: {
+    height: number;
+    width: number;
+  };
+};
+
+export type Attribute = 'OBJECTID' | 'string' | 'double' | 'Int32' | string;
+
+export type BatchTable = {
+  [key: string]: any[];
+};
+
+export type TileContent = {
+  _tile: TileHeader;
+  _tileset: Tileset3D;
+};
+
+export type B3DMContent = {
+  batchTableJson?: {[key: string]: any[]};
+  byteLength: number;
+  byteOffset: number;
+  cartesianModelMatrix: Matrix4;
+  cartesianOrigin: Vector3;
+  cartographicOrigin: Vector3;
+  featureIds?: number[] | null;
+  featureTableBinary?: Uint8Array;
+  featuretableJson?: {[key: string]: any};
+  gltf?: GLTFObject;
+};
+
+// TODO Add more properties here....
+type GLTFObject = {
```

Comment:
Can we use types from the `gltf` module?

Reply:
This object doesn't have the same structure as in `gltf` module. It has advanced structure. Also a lot of internal types of `GLTFObject`  are taken `from` gltf module.